Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.cpp

```diff
@@ -71,14 +71,12 @@ static void createAndBindCompositorBuffer(GL& gl, WebXRExternalRenderbuffer& buf
     LOG(XR, "WebXROpaqueFramebuffer::createAndBindCompositorBuffer(): created and bound external image to renderbuffer");
 }
 
-static GL::ExternalImageSource makeExternalImageSource(const PlatformXR::FrameData::ExternalTexture& imageSource, WebCore::IntSize size)
+static GL::ExternalImageSource makeExternalImageSource(PlatformXR::FrameData::ExternalTexture& imageSource, WebCore::IntSize size)
 {
     return GraphicsContextGLExternalImageSource {
-        .fds = imageSource.fds.map([](const UnixFileDescriptor& fd) {
-            return fd.duplicate();
-        }),
-        .strides = imageSource.strides,
-        .offsets = imageSource.offsets,
+        .fds = WTFMove(imageSource.fds),
+        .strides = WTFMove(imageSource.strides),
+        .offsets = WTFMove(imageSource.offsets),
```

Comment:
While this moves instead of duplicating, is it OK from the point of view of the caller? 

Are callers passing in an `imageSource` prepare to handle moved-out resource? Because we're not doing exchange: the old values are now in a valid, unspecified value if I understand it correctly.

Reply:
Yes, we just check if the fds vector is empty to decide and it's empty after the move

Reply:
Because in the case of Vector move always makes the moved one empty.